NSS-Day-Cohort-49/tabloid-mvc-tabloid-mvc-cassettes
# Mat comments

#### Changes Made
1. Added comment controller, repository, viewmodels, and views.
1. Refactored post add / edit to work with updates to categories and include category name in selection
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout mat-comments
    ```
2. Open a new Terminal tab (?T) and navigate to the server directory.
3. Test app functionality.

Post Refactor
    > Test post add / edit and (you should be able to select categories from a drop-down list by name)
#28 
    > Navigate to post details. You should have an option to leave a comment (single speech bubble)
    > Selecting this should navigate you to a form where you can add a subject and a comment. Saving the comment should direct you to list of comments for that post
#29
    > List of comments should be ordered with most recent on top. Title of post at top of page. Should display subject, content, author's display name, and creation date for each. You can also get to comment list by the icon with multiple speech bubbles from post details.
#27 
    > Given you are the user who made a comment, you should be able to delete the comment. The user should see the updated comment list.
#26
    > Given you are the user who made a comment, you should be able to edit the comment. If a user chooses not to edit, they should be able to redirect back to the list page. When the user saves edits, the comment should be updated and the user should be navigated to the comment details page.

4. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.
